src/Infrastructure/LeanCode.Serialization/JsonConverters/JsonLaxDateOnlyConverter.cs

```diff
@@ -1,18 +1,47 @@
+using System.Buffers;
 using System.Globalization;
+using System.Text;
 using System.Text.Json;
 using System.Text.Json.Serialization;
 
 namespace LeanCode.Serialization;
 
 public class JsonLaxDateOnlyConverter : JsonConverter<DateOnly>
 {
+    private static readonly UTF8Encoding encoder = new(false);
+    private const int maxBufferSize = 32;
+    private readonly string format = "yyyy-MM-dd";
+
     public override DateOnly Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
     {
-        return reader.GetString() is string s
-            ? DateOnly.FromDateTime(DateTime.Parse(s, CultureInfo.InvariantCulture))
-            : default;
+        Span<char> source = stackalloc char[0];
+
+        if(reader.HasValueSequence)
+        {
+            var valueLength = (int)reader.ValueSequence.Length;
+            Span<char> buffer = ArrayPool<char>.Shared.Rent(valueLength);
```

Comment:
You must return memory you rent from the array pool.

It would be easier if you just ensured the length does not exceed some reasonable value and always stackalloc'd the buffer. Try the following instead (pick one `#if` path):

```csharp
var length = reader.HasValueSequence ? (int)reader.ValueSeqence.Length : reader.ValueSpan.Length;

#if THROW_EXCEPTION
if (length > MaxBufferSize)
{
    throw new JsonException($"Input too long ({length})");
}

Span<char> buffer = stackalloc char[length];
#else
Span<char> buffer = length > MaxBufferSize ? new char[length] : stackalloc char[length];
#endif
var count = reader.CopyString(buffer);

if (DateTime.TryParse(buffer[..count], out var dt))
{
    return DateOnly.FromDateTime(dt);
}
else
{
    throw new JsonException("Failed to parse input, invalid format.");
}
```